main/githooks.py

```diff
@@ -411,6 +442,20 @@ def check_filename(filepath):
     return 0
 
 
+class TestCheckFileName(unittest.TestCase):
+    def test_various_strings(self):
+        def _test(input, output):
+            self.assertEqual(output, check_filename(input))
+        _test('good/some.txt', 0)
+        _test('bad/illegal/star*star.txt', 1)
+        _test('bad/reserved/device/con.txt', 1)
+        _test('bad/end/period.txt.', 1)
+        _test('bad/end/space.txt ', 1)
+        _test('bad/ascii/你好.txt', 1)
```

Comment:
Sigh: proper handling of unicode is still far from a solved problem

Reply:
This is a good check. In my last workplace we had some test files with non ascii file names and some systems handled them in unexpected way.